File: alafia-web-application/src/app/model/booking.ts
import {Client} from './client';

export class Booking {
  id: string;
  client: Client;
  diners: Client[];

  constructor(id: string, client: Client, diners: Client[]) {
    this.id = id;
    this.client = client;
    this.diners = diners;
  }
}

File: alafia-server/src/main/java/es/alafia/server/model/Booking.java
package es.alafia.server.model;

import lombok.Builder;
import lombok.Data;
import org.springframework.data.annotation.Id;

import java.util.List;

@Data
@Builder
public class Booking {

    @Id
    private String id;

    private Client client;

    private List<Client> diners;
}
